### PR TITLE
Document `OS.execute()` limitations on Android

### LINKED ISSUE #77639: the OS.execute () function on the Android platform
### Godot version

4.0.1 stable

### System information

HUAWEI Nova 5 Pro

### Issue description

When I want to run the OS.execute () function on the Android platform, [**OS. execute ("dumpsys", ["location"], output)**], it is clear that I have already opened the positioning authority of the application. But the result is [**permission denial: can't dump location manager service from PID = 27381, uid = 10192 due to missing Android. permission.dump permission**].

_How can I execute **“adb shell dumpsys location”**?_

### Steps to reproduce

略

### Minimal reproduction project

略

### PR DESCRIPTION
**Pull request by Calinou**
- This closes https://github.com/godotengine/godot/issues/77639.

### CODE REVIEW

File: doc/classes/OS.xml [hunk 1]
```diff
@@ -125,6 +125,7 @@
 				[b]Note:[/b] To execute a PowerShell built-in command, specify [code]powershell.exe[/code] in [param path], [code]-Command[/code] as the first argument, and the desired command as the second argument.
 				[b]Note:[/b] To execute a Unix shell built-in command, specify shell executable name in [param path], [code]-c[/code] as the first argument, and the desired command as the second argument.
 				[b]Note:[/b] On macOS, sandboxed applications are limited to run only embedded helper executables, specified during export.
+				[b]Note:[/b] on Android, system commands such as [code]dumpsys[/code] can only be run on a rooted device.
```
Comment (AThousandShips) [inline doc/classes/OS.xml:128]:
```suggestion
				[b]Note:[/b] On Android, system commands such as [code]dumpsys[/code] can only be run on a rooted device.
```

**Review by AThousandShips — APPROVED** [review]

**Review by Mickeon — APPROVED** [review]
This is the identifier with the most notes across the entire class reference.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.
Cherry-picked for 4.1.4.

**Pull request merged**